src/shared/vscode/localize.d.ts

```diff
@@ -0,0 +1,22 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+import { NominalType } from '../../shared/utilities/tsUtils'
+
+type PatchOverloadedReturnType<T, R> = T extends {
```

Comment:
This type only patches the first two overloaded signatures it finds. It's extremely difficult (if it's even possible) to write a type alias to parse an arbitrary number of call signatures (trust me, I tried...)